ibis/tests/benchmarks/test_benchmarks.py

```diff
@@ -753,3 +753,80 @@ def parse_many(types):
 
     types = ["VARCHAR", "INTEGER", "DOUBLE", "BIGINT"] * 1000
     benchmark(parse_many, types)
+
+
+def generate_asof_join_benchmark_data(
+    nprices: int, ntransactions: int
+) -> tuple[pd.DataFrame, pd.DataFrame]:
+    # Generate synthetic time data for prices
+    prices_times = pd.date_range(start="09:00", periods=nprices, freq="T")
+
+    # Define potential transaction times (every 00 or 30 seconds)
+    transaction_times = pd.date_range(
+        start="09:00", periods=ntransactions * 2, freq="30S"
+    )
+
+    # Sample from the potential transaction times for transactions
+    times_transactions = np.random.choice(
+        transaction_times, size=ntransactions, replace=False
+    )
+
+    # Generate synthetic price data
+    prices = (
+        102.5 + np.sin(np.linspace(0, nprices, nprices) / 100) * 2.0
+    )  # Sin wave around 102.5 with amplitude of 2
+
+    # Generate synthetic transaction data
+    shares = np.random.randint(5, 20, size=ntransactions)
+
+    # Create DataFrames
+    prices_df = pd.DataFrame({"time": prices_times, "price": prices})
+    transactions_df = pd.DataFrame({"time": times_transactions, "shares": shares})
+
+    # Sort transactions_df by time
+    return prices_df, transactions_df.sort_values(by="time")
+
+
+@pytest.fixture
+def asof_sql() -> str:
+    return """
+    SELECT t.time, t.shares, p.price
+    FROM transactions t
+    ASOF JOIN prices p
+      ON t.time >= p.time
+    """
+
+
+@pytest.fixture(scope="session")
+def ddb_asof_join(tmp_path_factory):
+    duckdb = pytest.importorskip("duckdb")
+
+    ntransactions = 100_000
+    nprices = 5 * ntransactions
+    prices_df, transactions_df = generate_asof_join_benchmark_data(
+        nprices, ntransactions
+    )
+    path = str(tmp_path_factory.mktemp("duckdb_asof_join") / "data.ddb")
+    with duckdb.connect(path) as con:
+        con.execute("CREATE TABLE transactions AS SELECT * FROM transactions_df")
+        con.execute("CREATE TABLE prices AS SELECT * FROM prices_df")
+    return path
+
+
+def test_asof_duckdb(benchmark, asof_sql, ddb_asof_join) -> None:
+    # yes, we're benchmarking duckdb here, not ibis
+    #
+    # we do this to get a baseline for comparison
```

Comment:
Is there a way where the benchmark result could be a ratio of `ibis-time / duckdb-time`? That way if the benchmark ratio number increases we'd see that as an immediate regression. I know airspeed velocity supports arbitrary benchmark metrics, not sure about pytest-benchmark.

Reply:
I don't see anything in the documentation around this.